src/smoothscroll.js

```diff
@@ -0,0 +1,279 @@
+(function(WIN, DOC, undefined) {
+  'use strict';
+
+  /*
+   * aliases
+   * WIN: window global object
+   * DOC: document
+   * undefined: undefined
+   */
+
+  // polyfill
+  function polyfill() {
+    // return when scrollBehavior interface is supported
+    if ('scrollBehavior' in DOC.documentElement.style) {
+      return;
+    }
+
+    /*
+     * globals
+     */
+    var Element = WIN.HTMLElement || WIN.Element;
+    var SCROLL_TIME = 468;
+
+    /*
+     * object gathering original scroll methods
+     */
+    var original = {
+      scroll: WIN.scroll || WIN.scrollTo,
+      scrollBy: WIN.scrollBy,
+      scrollIntoView: Element.prototype.scrollIntoView
+    };
+
+    /**
+     * changes scroll position inside an element
+     * @method scrollElement
+     * @param {Number} x
+     * @param {Number} y
+     */
+    function scrollElement(x, y) {
+      this.scrollLeft = x;
+      this.scrollTop = y;
+    }
+
+    /**
+     * get actual time in milliseconds
+     * @method now
+     * @returns {Number}
+     */
+    function now() {
```

Comment:
nit: to avoid checking for this each time let’s precompute which function now points to. The legit version of:

```
const now = WIN.performance ? WIN.performance.now : Date.now
```